### DISCUSSION TITLE
Security recommendations doc needed

### DISCUSSION THREAD

**Discussion by dskrvk** [requarks/wiki#3280] [Error / Bug Report]
**Describe the bug**
The documentation wiki doesn't have a clear set of guidelines for deploying the wiki securely. The default configuration may be insecure on some platforms.

**To Reproduce**
One of the issues is GraphQL queries returning an error stack trace in the response in the default wiki config, e.g. try running this query in the GraphQL console:
```
{
  authentication {
    strategies(isEnabled: true) {
      keys
      title
      useForm
      icon
      color
      selfRegistration
      __typename
    }
    __typename
  }
}
```
The response includes a stack trace containing local paths on the server running the wiki.

Since the wiki can be deployed using different means and platforms, ensuring the deployment is secure involves both
1. providing good default configurations for the main platforms mentioned on https://docs.requarks.io/install and
2. listing some things to manually verify in case a custom deployment method is used.

**Expected behavior**
1. In general, a docs page describing a checklist for a secure deployment.
2. For the specific issue described above, the page should mention that `NODE_ENV=production` needs to be set when running the wiki.

**Host Info (please complete the following information):**
 - OS: Amazon Linux (linux) 2018.03 x64
 - Wiki.js version: 2.3.81
 - Database engine: postgres 10

**Additional context**

> To disable stacktraces for production, pass debug: false to the Apollo Server constructor or set the NODE_ENV environment variable to 'production' or 'test'.
* https://www.apollographql.com/docs/apollo-server/data/errors/#default-information
* https://expressjs.com/en/advanced/best-practice-performance.html#set-node_env-to-production

**Comment by ghost** [comment]
You haven't provided the required info about your host! (OS, Wiki.js version, Database engine)
